app/models/cohort.rb

```diff
@@ -35,11 +35,22 @@ def class_dates_until(last_date)
     (start_date..last_date).select { |date| !date.saturday? && !date.sunday? }
   end
 
+  def internships_sorted_by_interest(current_student)
+    self.internships.sort_by do |internship|
+      rating = current_student.find_rating(internship)
+      if rating
+        rating.interest
+      else
+        '0'
+      end
+    end
+  end
+
 private
 
   def import_code_reviews
     unless @importing_cohort_id.blank?
-      self.code_reviews = Cohort.find(@importing_cohort_id).deep_clone(include: { code_reviews: :objectives }).code_reviews
+      code_reviews = Cohort.find(@importing_cohort_id).deep_clone(include: { code_reviews: :objectives }).code_reviews
```

Comment:
Oh wow, I can't believe I did that. I was trying to remove the self from line 39.

` self.internships.sort_by do |internship|` 

This is also a lesson in why we have automated testing, to save us from ourselves :+1: